tests/ui/drop_forget_ref.rs

```diff
@@ -55,3 +55,18 @@ fn test_similarly_named_function() {
     forget(&SomeStruct); //OK; call to unrelated function which happens to have the same name
     std::mem::forget(&SomeStruct);
 }
+
+#[allow(dead_code)]
+fn test_owl_closure() -> Result<(), ()> {
```

Comment:
I'm not sure if this should be restricted only to owl results instead of all result types with either their ok or err types equal to `()`